rpcapi/votehandlers.go

```diff
@@ -25,11 +25,11 @@ func (r *RPCAPI) submitRawTx(request *api.APIrequest) (*api.APIresponse, error)
 		return nil, fmt.Errorf("no reply from vochain")
 	}
 	if res.Code != 0 {
-		return nil, fmt.Errorf("%s", res.Data)
+		return nil, fmt.Errorf("%s", string(res.Data))
```

Comment:
this is the correct change. the problem is that tendermint's HexBytes has a String method that does an uppercase hex encode. we don't want that - error strings are always plaintext and not binary, so we can return them directly.

*however*, note that this will mean that the client side must only do hex decode on `resp.Data` if *and only if* `resp.Code==0`. Otherwise, the response error message will be in `resp.Data` as a plain string, without hex encoding. Is that how the API is documented today? If not, and we merge this, we should also fix the API docs.

cc @brickpop @marcvelmer

Reply:
I'm not sure I'm following your comment @mvdan 

I think in case of res.Code != 0 we should `return nil, fmt.Errorf(hex.DecodeString(util.trimHex(res.Data)))`
The RPCAPI will take care of setting the string as a JSON error message for the client.

Reply:
res.Data holds bytes, not a hex string, so I'm pretty sure your code wouldn't work?

the problem is that res.Data is tendermint's `type HexBytes []byte`, whose String method encodes as hex. our previous code was calling that String method via `%s`. the new code prevents that by instead converting to `string`, which then gives us the non-encoded bytes as a string directly.

Reply:
I see, thanks.

Reply:
So lets merge this @mvdan @NateWilliams2 ?

Reply:
Yeah, I think so. What I said earlier about `resp.Data` being hex only some of the time is wrong, it's always hex - and it comes from tendermint. It's us who have to be careful to not encode as hex when returning errors. That's the fix Nate is doing here. LGTM :)